train_scripts/train_sana_sprint_diffusers.py

```diff
@@ -0,0 +1,1823 @@
+    logger.info(f"  Num examples = {len(train_dataset)}")
+    logger.info(f"  Num batches each epoch = {len(train_dataloader)}")
+    logger.info(f"  Num Epochs = {args.num_train_epochs}")
+    logger.info(f"  Instantaneous batch size per device = {args.train_batch_size}")
+    logger.info(f"  Total train batch size (w. parallel, distributed & accumulation) = {total_batch_size}")
+    logger.info(f"  Gradient Accumulation steps = {args.gradient_accumulation_steps}")
+    logger.info(f"  Total optimization steps = {args.max_train_steps}")
+    global_step = 0
+    first_epoch = 0
+
+    # Potentially load in the weights and states from a previous save
+    if args.resume_from_checkpoint:
+        if args.resume_from_checkpoint != "latest":
+            path = os.path.basename(args.resume_from_checkpoint)
+        else:
+            # Get the mos recent checkpoint
+            dirs = os.listdir(args.output_dir)
+            dirs = [d for d in dirs if d.startswith("checkpoint")]
+            dirs = sorted(dirs, key=lambda x: int(x.split("-")[1]))
+            path = dirs[-1] if len(dirs) > 0 else None
+
+        if path is None:
+            accelerator.print(
+                f"Checkpoint '{args.resume_from_checkpoint}' does not exist. Starting a new training run."
+            )
+            args.resume_from_checkpoint = None
+            initial_global_step = 0
+        else:
+            accelerator.print(f"Resuming from checkpoint {path}")
+            accelerator.load_state(os.path.join(args.output_dir, path))
+            global_step = int(path.split("-")[1])
+
+            initial_global_step = global_step
+            first_epoch = global_step // num_update_steps_per_epoch
+
+    else:
+        initial_global_step = 0
+
+    progress_bar = tqdm(
+        range(0, args.max_train_steps),
+        initial=initial_global_step,
+        desc="Steps",
+        # Only show the progress bar once on each machine.
+        disable=not accelerator.is_local_main_process,
+    )
+
+    phase = "G"
+    vae_config_scaling_factor = vae.config.scaling_factor
+    sigma_data = args.sigma_data
+    negative_prompt = [""] * args.train_batch_size
+    negative_prompt_embeds, negative_prompt_attention_mask = text_encoding_pipeline.encode_prompt(
+        prompt=negative_prompt,
+        complex_human_instruction=False,
+        do_classifier_free_guidance=False,
+    )
+
+    for epoch in range(first_epoch, args.num_train_epochs):
+        transformer.train()
+        disc.train()
+
+        for step, batch in enumerate(train_dataloader):
+            # text encoding
+            prompts = batch["prompts"]
+            text_encoding_pipeline = text_encoding_pipeline.to("cuda")
+            with torch.no_grad():
+                prompt_embeds, prompt_attention_mask, _, _ = text_encoding_pipeline.encode_prompt(
+                    prompts, complex_human_instruction=complex_human_instruction, do_classifier_free_guidance=False
+                )
```

Comment:
Let's bring the `text_encoding_pipeline` to CPU once the prompt embeddings are computed?